.github/scripts/compare-images-and-pdfs.py

```
@@ -0,0 +1,493 @@
+
+        Returns True when the comment was posted/updated successfully and
+        False when the API rejected the request because the token is not
+        permitted to write comments (this happens for pull requests opened
+        from forks when the workflow runs with a read-only GITHUB_TOKEN).
+        Any other failure is raised so it is surfaced to the caller.
+        """
+        url = f'{self.base_url}/issues/{self.pr_number}/comments'
+
+        # Check if we already posted a comment
+        existing_comments_url = f'{self.base_url}/issues/{self.pr_number}/comments'
+        response = requests.get(existing_comments_url, headers=self.headers)
```

Ruff (0.15.20)

[error] Probable use of `requests` call without timeout

(S113)